sdk/iOS/src/MSCoreDataStore.m

```diff
@@ -214,7 +207,6 @@ -(MSSyncContextReadResult *)readWithQuery:(MSQuery *)query orError:(NSError *__a
         fr.entity = entity;
         fr.predicate = query.predicate;
         fr.sortDescriptors = query.orderBy;
-        fr.resultType = NSDictionaryResultType;
```

Comment:
I could not find any related documentation and I don't see an easy way to measure it. Another option which I considered, but chose not to use, was to stick to NSDictionaryResultType and check the returned NSDictionary for missing properties. 

However, it might actually make sense to take that approach. We would need to check for missing properties only if the property count of an item does not match the number of properties requested. That way we do extra work (potentially O(n^2) where n == no. of properties) only if an item has one or more null properties.

What do you think?